esp-bootloader-esp-idf/Cargo.toml

```diff
@@ -1,6 +1,6 @@
 [package]
 name = "esp-bootloader-esp-idf"
-version = "0.1.0"
+version = "0.0.0"
```

Comment:
I'm resetting this, otherwise the release process would like to release `0.2` immediately. This is a bit of a side effect, the tool doesn't understand that a crate hasn't been published yet. Is this a concern, should we prepare for unreleased crates?

Reply:
I think it's likely we'll add more crates to our ecosystem, so I think it's something we should handle. I think we can maybe decide that a `0.0.0` crate is treated by the tooling as unreleased (maybe there is a better sentinel value we can use that cargo will accept).

Reply:
As far as I know cargo has no issues with 0.0.0. ([example crate](https://crates.io/crates/hebrew_accents)). The tooling doesn't actually have any issues finalizing a version bump for them, it's just the initial version bump that is somewhat special - at least so far this is what I've been seeing.

Reply:
There is `publish = false` in the manifest format ([link](https://doc.rust-lang.org/cargo/reference/manifest.html#the-publish-field))

Reply:
Thanks, that's something I missed and we should definitely use it. We already maintain a list of unpublished crates in the code, but this would actually make them unpublished from the PoV of cargo.

Reply:
> There is `publish = false` in the manifest format ([link](https://doc.rust-lang.org/cargo/reference/manifest.html#the-publish-field))

Resolved via https://github.com/esp-rs/esp-hal/pull/3507.

I think we can settle for `0.0.0` as the start value for any package.